### PR TITLE
Release 1.3.2 0.9.14

### PR DESCRIPTION
**Pull request by ruKurz**
This releases the latest fixes to our production:

- Added Microsoft Text Translator as service to auto translate
- Improved translated message cache
- Fixed production deployment steps in circleCI

### CODE REVIEW

File: app/autotranslate/client/lib/autotranslate.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ export const AutoTranslate = {
 		Tracker.autorun(() => {
 			Subscriptions.find().observeChanges({
 				changed: (id, fields) => {
-					if (fields.hasOwnProperty('autoTranslate')) {
+					if (fields.hasOwnProperty('autoTranslate') || fields.hasOwnProperty('autoTranslateLanguage')) {
```
Comment (mrsimpson) [inline app/autotranslate/client/lib/autotranslate.js]: @vickyokrm this looks as if the change of language was not observed anymore. Why did you remove this?

File: app/autotranslate/client/views/autoTranslateFlexTab.js [hunk 2]
```diff
@@ -99,7 +99,7 @@ Template.autoTranslateFlexTab.onCreated(function() {
 	this.saveSetting = () => {
 		const field = this.editing.get();
 		const subscription = Subscriptions.findOne({ rid: this.rid, 'u._id': Meteor.userId() });
-		const previousLanguage = subscription.autoTranslateLanguage;
+		const previousLanguage = subscription && subscription.autoTranslateLanguage;
```
Comment (mrsimpson) [inline app/autotranslate/client/views/autoTranslateFlexTab.js]: How could you toggle language without being subscribed to the room?

File: app/autotranslate/server/deeplTranslate.js [hunk 3]
```diff
@@ -150,7 +150,7 @@ class DeeplAutoTranslate extends AutoTranslate {
 					translations[language] = this.deTokenize(Object.assign({}, message, { msg: translatedText }));
 				}
 			} catch (e) {
-				SystemLogger.error('Error translating message', e);
+				logger.deepl.error('Error translating message', e);
```
Comment (mrsimpson) [inline app/autotranslate/server/deeplTranslate.js]: Interesting! what's the effect of `.deepl.`?

File: app/ui-utils/client/lib/renderMessageBody.js [hunk 4]
```diff
@@ -32,4 +33,4 @@ export const renderMessageBody = mem((message) => {
 
 	return (Array.isArray(tokens) ? tokens.reverse() : [])
 		.reduce((html, { token, text }) => html.replace(token, () => text), html);
-}, 500, ({ _id, _updatedAt }) => _id && _updatedAt && _id + _updatedAt);
+}, 500, (message) => md5(JSON.stringify(message)));
```
Comment (mrsimpson) [inline app/ui-utils/client/lib/renderMessageBody.js]: Does this make an editing of the message re-render it? Or what's the motivation?

File: app/autotranslate/client/lib/autotranslate.js (hunk 1 above)
Comment (vickyokrm) [reply app/autotranslate/client/lib/autotranslate.js]: It is observed.

File: app/autotranslate/client/views/autoTranslateFlexTab.js (hunk 2 above)
Comment (jschirrmacher) [reply app/autotranslate/client/views/autoTranslateFlexTab.js]: You can't, currently. That would be another fix we should do, or at least hide the translation panel / button if not subscribed.

File: app/ui-utils/client/lib/renderMessageBody.js (hunk 4 above)
Comment (jschirrmacher) [reply app/ui-utils/client/lib/renderMessageBody.js]: Yes, editing will re-renders, but editing would also change `_updatedAt`, so that the message was re-rendered with the previous code as well. The new code also re-renders if the translation changes.

However, instead of using `md5` library, I would prefer to just use `crypto`, which is already included in Node.js and also has the option to generate a md5 hash.

Comment (jschirrmacher) [reply app/ui-utils/client/lib/renderMessageBody.js]: Sth. like

    crypto.createHash('md5').update(message).digest("hex")

File: app/autotranslate/server/deeplTranslate.js (hunk 3 above)
Comment (vickyokrm) [reply app/autotranslate/server/deeplTranslate.js]: using sections will print the logger name on the console. It easy to identify from which provider the exception is raised.

File: app/autotranslate/client/lib/autotranslate.js (hunk 1 above)
Comment (mrsimpson) [reply app/autotranslate/client/lib/autotranslate.js]: sorry, wrong change mode

File: app/autotranslate/client/views/autoTranslateFlexTab.js (hunk 2 above)
Comment (mrsimpson) [reply app/autotranslate/client/views/autoTranslateFlexTab.js]: ok, doesn't hurt anyway

File: app/autotranslate/server/deeplTranslate.js (hunk 3 above)
Comment (mrsimpson) [reply app/autotranslate/server/deeplTranslate.js]: see `app/autotranslate/server/logger.js`

File: app/ui-utils/client/lib/renderMessageBody.js (hunk 4 above)
Comment (mrsimpson) [reply app/ui-utils/client/lib/renderMessageBody.js]:
```javascript

const createHashKey = function(o){
    return crypto.createHash('md5').update(String(JSON.stringify(o))).digest("hex")
}
```
and then use only the properties relevant to caching
```javascript
({_id, updatedAt, translations}) => createHashKey(...arguments)
```

**Comment by ruKurz** [comment]
> Also, I noticed there's neither a release branch nor tag nor release-notes of version-bump. Please add that, @ruKurz
Unfortunately one can not change the source branch of a PR. So by creating a new release branch, which is absolute necessary w'll lose your valuable comments ;-(

**Comment by jschirrmacher** [comment]
You can always [rebase to another branch](https://git-scm.com/book/de/v1/Git-Branching-Rebasing)

**Comment by jschirrmacher** [comment]
Would be great to have #652 and #653 also in the next release.

**Comment by ruKurz** [comment]
> > Also, I noticed there's neither a release branch nor tag nor release-notes of version-bump. Please add that, @ruKurz
> 
> Unfortunately one can not change the source branch of a PR. So by creating a new release branch, which is absolute necessary w'll lose your valuable comments ;-(

I started a new PR: https://github.com/assistify/Rocket.Chat/pull/654
